pkg/crc/oc/clusteroperator.go

```diff
@@ -9,7 +9,7 @@ import (
 
 var ignoreClusterOperators = []string{"monitoring", "machine-config", "marketplace"}
 
-type ClusterOperator struct {
+type K8sResource struct {
```

Comment:
`K8sResource` is kind of a very generalize way and can have anything (like pod, deployment, rs ..etc.) , If I really want to change name then I would always include which type of k8s resource is this one.

Reply:
This `ClusterOperator` structure is generic enough to match what is need for `Csr`, hence the more generic name. I don't know how to call it otherwise? I can change this to
```
type Csr ClusterOperator
type ClusterOperator struct {
 ...
}
```
I suspect we might be reinventing struct already defined in https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go for example, might be worth using the kubernetes API directly at some point rather than going through `oc`.

Reply:
> I suspect we might be reinventing struct already defined in https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go for example, might be worth using the kubernetes API directly at some point rather than going through oc

I think, in past we did try that and found out it is k8s dep is too heavy to include in the vendor directory than just parsing some of the `oc` command output.

Reply:
> This ClusterOperator structure is generic enough to match what is need for Csr, hence the more generic name. I don't know how to call it otherwise? I can change this to

What's wrong with following
```
var csrs K8sResource
```
to 
```
var csrs ClusterOperator
```

Reply:
> I think, in past we did try that and found out it is k8s dep is too heavy to include in the vendor directory than just parsing some of the `oc` command output.

This means instead of having k8s in vendor/, we have k8s hidden in the `oc` binary, I don't think this really is different, just hidden :)

Reply:
> What's wrong with following
> 
> ```
> var csrs K8sResource
> ```
> 
> to
> 
> ```
> var csrs ClusterOperator
> ```

Why are we using something named `ClusterOperator` in order to parse `oc get csr -o yaml` output? This would be confusing/misleading.

Reply:
> This means instead of having k8s in vendor/, we have k8s hidden in the oc binary, I don't think this really is different, just hidden :)

This is different `oc` binary we anyway required to all the kind of operation and also after the cluster is running, what you are suggesting here is completely drop  oc commands an just use the k8s apis (even with login or any kind of stuff) this is what odo do atm. I am not sure how much effort will it take for us to replace all the oc commands with the API endpoints.

Reply:
> Why are we using something named ClusterOperator in order to parse oc get csr -o yaml output? This would be confusing/misleading.

yes, I read again so  k8sresource struct same here for clusteroperator and csr so it is just reused 👍